File: src/types/cursor.ts
export interface Package {
    id: number;
    name: string;
    description: string;
    price: number;
}

export interface CursorType {
    id: number;
    name: string;
    description: string;
}

export interface Cursor {
    id: number;
    packageId: number;
    cursorTypeId: number;
    cursorName: string;
    pathToIcon: string;
    package?: Package;
    type?: CursorType;
} 

File: SeniorCursor.API/Models/Cursor.cs
namespace SeniorCursor.API.Models;

public class Cursor
{
    public int Id { get; set; }
    public int PackageId { get; set; }
    public int CursorTypeId { get; set; }
    public string CursorName { get; set; } = string.Empty;
    public string PathToIcon { get; set; } = string.Empty;
    public Package Package { get; set; } = null!;
    public CursorType Type { get; set; } = null!;
} 